src/metabase/api/table.clj

```diff
@@ -37,22 +41,40 @@
   (-> (api/read-check Table id)
       (hydrate :db :pk_field)))
 
+(defn- visible-state?
+  "only the nil state is considered visible."
```

Comment:
Since this function ends in a `?` it sort of implies it returns a truthy or falsy value. Maybe a name like `visibility-type->visibility-state` would be more appropriate or even just making this return `true` instead of `:show` and `false` instead of `hide` would make it a lot simpler